text/0014-ai-prompt-api.md

```diff
@@ -0,0 +1,176 @@
+# RFC Template
+
+- Start Date: 2025-04-09
+- RFC PR: [electron/rfcs#14](https://github.com/electron/rfcs/pull/14)
+- Electron Issues:
+- Reference Implementation:
+- Status: **Proposed**
+
+# Prompt API support
+
+## Summary
+
+The Prompt API is part of [Built-in AI](https://developer.chrome.com/docs/ai/built-in)
+capabilities in Chromium based browsers. With the Prompt API developers can send
+natural language requests from the renderer for local processing.  This proposal adds
+support for this API in Electron.
+
+## Motivation
+
+Built in AI features are being implemented in Chromium based browsers.  Electron should
+also support these features.
+
+## Guide-level explanation
+
+The Prompt API can be used by Electron app developers by defining a local AI handler
+script via `Session.registerLocalAIHandlerScript` and a new module `localAIHandler`.
+The handler script will run in a [utilityProcess](https://www.electronjs.org/docs/latest/api/utility-process)
+and will allow app developers to proxy Built-in API calls to a local LLM implementation of
+their choice.
+
+For the initial implementation, the local AI handler will only support the prompt API,
+but the long term intention is that this handler would also support other Built-in AI APIs.
+
+### Session
+
+`ses.registerLocalAIHandlerScript(options)`
+* `options` Object
+  * `filePath` string - Path of the script file. Must be an absolute path.
+
+Returns `string` - The ID of the registered local AI handler script.
+
+`ses.unregisterLocalAIHandlerScript(id)`
+
+* `id` string - local AI handler script ID
+
+Unregisters the specified local AI handler script and terminates the associated
```

Comment:
Will this call the `destroy()` method on any prompt API handlers,  or does that only happen if the prompt instance is destroyed?

Are there any other considerations we should make for how the utility process should handle shutdown, or is that already covered by existing utility process APIs?